### ISSUE TITLE
Powertop claims Nvidia card at 100% usage

### ISSUE THREAD

**Issue by michaeldfallen** [Bumblebee-Project/Bumblebee#705]
I've been grappling with Bumblebee and Nvidia-prime for a while and while I have it working I'm not seeing the power savings I would have expected. When I run Powertop, in the devices tab, it claims:
```
Usage    Device

100%     PCI Device: NVIDIA Corporation GM108M [GeForce 940M]|
```
If bumblebee was doing it's job correctly should that not be at 0%? Is the card still drawing power despite not being utilised? Should I be setting `Runtime PM for ...` to Good?

-----------------------

My setup (for background):

I've been documenting everything in [my puppet repo](https://github.com/michaeldfallen/rusty-boxen). Specifically [optimus.pp](https://github.com/michaeldfallen/rusty-boxen/blob/master/modules/ubuntu/manifests/optimus.pp) shows the pacakges I've installed.

I have:
- Nvidia-prime - set the `Intel` profile so that the Desktop environment (Unity) runs on the intel card
- Nvidia-352 - latest driver for my card
- Bumblebee - with Nvidia-prime handling the "don't use the nvidia card" part bumblebees optirun lets me opt in to using it for games etc.
- /etc/bumblebee/bumblebee.conf set with Nvidia-352 as the driver, kernel-driver and location

All that means that I can successfully run Unity on the Intel card and switch to the Nvidia card for games etc using `primusrun` or `optirun`.

**Comment by michaeldfallen** [comment]
BTW, if I set `Runtime PM...` to Good in powertop and refresh my xsession I'm seeing 0% usage until I run an app using optirun. Does that mean I would be getting better power savings? Why isn't the device going back to 0% after the optiran app closes?

**Comment by michaeldfallen** [comment]
btw: `cat /proc/acpi/bbswitch` returns `0000:04:00.0 ON` even when the usage shows 0% in powertop.

**Comment by ArchangeGabriel** [comment]
Your issue here is probably bumblebee having trouble to unload the nvidia modules, so that bbswitch can’t deactivate it. We’re working on it, news soon.

**Comment by michaeldfallen** [comment]
Thanks for chasing this, I had forgot I opened this. 

https://github.com/michaeldfallen/rusty-boxen/commit/cc521b1efba0315f00cab0dbc5589128fb92fe0f from my boxen shows the fix. I used a [bbswitch module hacked by smunaut to support the T450s]( https://github.com/smunaut/bbswitch/tree/hack-t440s) as documented in Bumblebee-Project/bbswitch#112 to add proper full support for my 940M.

Now on boot bbswitch successfully powers down the card (it goes into Rev FF) and when I use `optirun` or `primusrun` the card powers on as expected. Occasionally it doesn't power off properly after an `optirun` or `primusrun` as the Nvidia kernel modules aren't unloaded correctly but I can manually unload them and trigger `sudo tee /proc/acpi/bbswitch <<<OFF` to properly power it off.
